pkg/storage/export/export/export.go

```diff
@@ -416,103 +524,194 @@ func (ctrl *VMExportController) updateVMExport(vmExport *exportv1.VirtualMachine
 	if err != nil {
-		return 0, err
-	} else if exists {
+		return nil, false, "", err
+	}
+	if exists {
 		if vmSnapshot.Status.ReadyToUse != nil && *vmSnapshot.Status.ReadyToUse {
-			pvcs, restoreableSnapshots, err = ctrl.handlePVCsForVirtualMachineSnapshot(vmExport, vmSnapshot)
+			pvcs, restoreableSnapshots, err := ctrl.handlePVCsForVirtualMachineSnapshot(vmExport, vmSnapshot)
 			if err != nil {
-				return 0, err
+				return nil, false, "", err
+			}
+			if len(pvcs) == restoreableSnapshots && restoreableSnapshots > 0 {
+				return pvcs, true, "", nil
+			}
+			if restoreableSnapshots == 0 {
+				return nil, false, fmt.Sprintf("VirtualMachineSnapshot %s/%s does not contain any volume snapshots", vmExport.Namespace, vmExport.Spec.Source.Name), nil
 			}
+			return pvcs, false, "Not all PVCs have been successfully restored", nil
+		} else {
+			return nil, false, fmt.Sprintf("VirtualMachineSnapshot %s/%s is not ready to use", vmExport.Namespace, vmExport.Spec.Source.Name), nil
 		}
-	} // If snapshot doesn't exist or is not ready, it will fall through to updateVMExportVMSnapshotStatus because no pod will exist and no pvcs
+	}
+	return nil, false, fmt.Sprintf("VirtualMachineSnapshot %s/%s does not exist", vmExport.Namespace, vmExport.Spec.Source.Name), nil
+}
 
-	pod, exists, err := ctrl.getExporterPod(vmExport)
+func (ctrl *VMExportController) isSourceAvailableVM(vmExport *exportv1.VirtualMachineExport) (bool, string, error) {
+	vmi, exists, err := ctrl.getVmi(vmExport.Namespace, vmExport.Spec.Source.Name)
 	if err != nil {
-		return 0, err
-	} else if !exists {
-		if len(pvcs) == restoreableSnapshots && restoreableSnapshots > 0 {
-			pod, err = ctrl.createExporterPod(vmExport, pvcs)
-			if err != nil {
-				return 0, err
-			} else if pod == nil {
-				return 0, nil
-			}
+		return false, "", err
+	}
+	if exists {
+		return false, fmt.Sprintf("Virtual Machine %s/%s is running", vmi.Namespace, vmi.Name), nil
+	}
+	return true, "", nil
+}
 
-			if err := ctrl.getOrCreateCertSecret(vmExport, pod); err != nil {
-				return 0, err
-			}
+func (ctrl *VMExportController) getPVCFromSourceVM(vmExport *exportv1.VirtualMachineExport) ([]*corev1.PersistentVolumeClaim, bool, string, error) {
+	pvcs, err := ctrl.getPVCsFromVM(vmExport.Namespace, vmExport.Spec.Source.Name)
+	if err != nil {
+		return nil, false, "", err
+	}
+	sourceAvailable, availableMessage, err := ctrl.isSourceAvailableVM(vmExport)
+	if err != nil {
+		return nil, false, "", err
+	}
+	return pvcs, sourceAvailable, availableMessage, nil
+}
+
+func (ctrl *VMExportController) getPVCsFromVM(vmNamespace, vmName string) ([]*corev1.PersistentVolumeClaim, error) {
+	var pvcs []*corev1.PersistentVolumeClaim
+	vm, exists, err := ctrl.getVm(vmNamespace, vmName)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, nil
+	}
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		pvc, exists, err := ctrl.getPvc(vmNamespace, storagetypes.PVCNameFromVirtVolume(&volume))
```

Comment:
`PVCNameFromVirtVolume` may return ""

Reply:
Right which then will return !exists from ctrl.getPvc due to the blank name.